doozerlib/distgit.py

```diff
@@ -563,28 +563,31 @@ def push_image(self, tag_list, push_to_defaults, additional_registries=[], versi
 
                     mirror_cmd = 'oc image mirror {} {} --filename={}'.format(dr, insecure, push_config)
 
-                    for r in range(10):
-                        self.logger.info("Mirroring image [retry={}]".format(r))
-                        rc, out, err = exectools.cmd_gather(mirror_cmd)
-                        if rc == 0:
-                            break
-                        self.logger.info("Error mirroring image -- retrying in 60 seconds.\n{}".format(err))
-                        time.sleep(60)
-
-                    lstate = self.runtime.state[self.runtime.command] if self.runtime.command == 'images:push' else None
-
-                    if rc != 0:
-                        if lstate:
-                            state.record_image_fail(lstate, self.metadata, 'Build failure', self.runtime.logger)
-                        # Unable to push to registry
-                        raise IOError('Error pushing image: {}'.format(err))
+                    if dry_run:  # skip everything else if dry run
+                        continue
                     else:
```

Comment:
you don't need the else though, because the continue skips the rest... was trying to save you the massive whitespace change :)